### PR TITLE
Add s390x unit test job

### PR DESCRIPTION
**Pull request by Davo911**
**What this PR does / why we need it**:

Add a prow job to be able to run unit tests on a s390x worker.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by awels** [comment]
/lgtm
Once the matching CDI PR is in, we can make the lane voting.

File: github/ci/prow-deploy/files/jobs/kubevirt/containerized-data-importer/containerized-data-importer-presubmits.yaml [hunk 1]
```diff
@@ -31,6 +31,37 @@ presubmits:
           resources:
             requests:
               memory: "8Gi"
+  - name: pull-cdi-unit-test-s390x
+    skip_branches:
+      - release-v1.38
+      - release-v1.43
+      - release-v1.49
+    cluster: s390x-virtual-workloads #
```
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/containerized-data-importer/containerized-data-importer-presubmits.yaml:39]: Could we use the same node that runs the kubevirt unit tests? 

https://github.com/kubevirt/project-infra/blob/15d3b253c6484d75f396700b16093a58a9668ae0/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml#L422C14-L422C32

File: github/ci/prow-deploy/files/jobs/kubevirt/containerized-data-importer/containerized-data-importer-presubmits.yaml [hunk 2]
```diff
@@ -31,6 +31,37 @@ presubmits:
           resources:
             requests:
               memory: "8Gi"
+  - name: pull-cdi-unit-test-s390x
+    skip_branches:
+      - release-v1.38
```
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/containerized-data-importer/containerized-data-importer-presubmits.yaml:36]: Does it make sense to use a similar regex to skip release branches? 

https://github.com/kubevirt/project-infra/blob/15d3b253c6484d75f396700b16093a58a9668ae0/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml#L432

**Review by brianmcarey — COMMENTED** [review]
Thanks @Davo911 for raising this PR - just a couple of inline comments.

File: github/ci/prow-deploy/files/jobs/kubevirt/containerized-data-importer/containerized-data-importer-presubmits.yaml [hunk 3]
```diff
@@ -31,6 +31,35 @@ presubmits:
           resources:
             requests:
               memory: "8Gi"
+  - name: pull-cdi-unit-test-s390x
+    skip_branches:
+      - release-\d+\.\d+
```
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/containerized-data-importer/containerized-data-importer-presubmits.yaml:36]:
```suggestion
      - release-v\d+\.\d+
```

I think the CDI branches include the `v`

Comment (awels) [reply github/ci/prow-deploy/files/jobs/kubevirt/containerized-data-importer/containerized-data-importer-presubmits.yaml:36]: They do

File: github/ci/prow-deploy/files/jobs/kubevirt/containerized-data-importer/containerized-data-importer-presubmits.yaml [hunk 4]
```diff
@@ -31,6 +31,35 @@ presubmits:
...
+      fork-per-release: "true"
+    always_run: true
+    optional: true
+    decorate: true
+    decoration_config:
+      timeout: 1h
+      grace_period: 5m
+    labels:
+      preset-podman-in-container-enabled: "true"
+      preset-docker-mirror-proxy: "true"
```
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/containerized-data-importer/containerized-data-importer-presubmits.yaml:48]: The image proxy is not available on the s390x cluster so this preset will have to be removed or the test pods will not be scheduled.

**Review by brianmcarey — COMMENTED** [review]
This looks good to me - just one more preset to remove and it should be good to go.

File: github/ci/prow-deploy/files/jobs/kubevirt/containerized-data-importer/containerized-data-importer-presubmits.yaml (hunk 4 above)
Comment (Davo911) [reply github/ci/prow-deploy/files/jobs/kubevirt/containerized-data-importer/containerized-data-importer-presubmits.yaml]: Done

**Review by brianmcarey — COMMENTED** [review]
/approve

This job config looks good to me and should do what is wanted. 

I haven't taken a look at the CDI enablement PRs but I guess there will be a s390x version of the CDI builder image that will be used by this job.
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/3552/rehearsal-pull-cdi-unit-test-s390x/1818191625902362624#1:build-log.txt%3A17

**Comment by cfilleke** [comment]
We'll need to configure the builder image to reside in the IBM cloud registry icr.io/kubevirt -- please let us know how to name it & invoke it.  I believe we'll need to provide an API token as an encrypted pull secret for it.

**Comment by cfilleke** [comment]
NB. It's not clear to me in the course of running these tests or configuring the CI/CD where & when the build-the-builder container is built, and how it is registered in the registry of choice: for s390x it should be registered in `icr.io/kubevirt` with an API key stored as an encrypted secret in the K8S cluster.

**Comment by Davo911** [comment]
Main issue here is the lack of bazel support for s390x. We'll be doing something similar to https://github.com/kubevirt/kubevirt/pull/11097

**Comment by Davo911** [comment]
Could you rerun the rehearsal by any chance? @brianmcarey

**Comment by Davo911** [comment]
I got the hint to maybe add the label `preset-bazel-unnested: "true"`, but cannot find any info regrading this.

**Comment by Davo911** [comment]
Now we've got the [multiarch builder available](https://quay.io/repository/kubevirt/kubevirt-cdi-bazel-builder?tab=tags&tag=latest), let's try the /rehearsal again.

**Comment by brianmcarey** [comment]
> Now we've got the [multiarch builder available](https://quay.io/repository/kubevirt/kubevirt-cdi-bazel-builder?tab=tags&tag=latest), let's try the /rehearsal again.

Yes sure - I am just waiting on the updated builder to get merged - rehearsals run against the main branch.

**Comment by Davo911** [comment]
Look like the test was killed midway through. Could be a flake?

**Comment by Davo911** [comment]
Ok not a flake and is also killed on my machine, when using the public emulation made builder.
When tested with my natively built builder image or running directly on the machine it works though.

**Comment by Davo911** [comment]
Actually every `make` command fails, when using the emulated builder, but when pulling it and manually running the tests inside, the unit tests succeed again...

**Comment by Davo911** [comment]
Closed this by accident, wasn't aware about the "Fixes" keyword in #3431.
Can this be reopened ? @brianmcarey @awels

**Comment by Davo911** [comment]
now this should work
/rehearse

**Review by brianmcarey — COMMENTED** [review]
/lgtm

Lane is optional and the rehearsal passed.

**Pull request merged**